app/api/threads/update_thread.feature

```diff
@@ -0,0 +1,386 @@
+    And the database has the following table 'threads':
+      | item_id | participant_id | status | helper_group_id | latest_update_at |
+    And the time now is "2022-01-01T00:00:00Z"
+
+  Scenario: Create a thread if it doesn't exist
+    Given I am the user with id "3"
+    And There is no thread with "item_id=1000,participant_id=3"
+    And I am in the group with id "100"
+    And I can request help to the group with id "100" on the item with id "1000"
+    When I send a PUT request to "/items/1000/participant/3/thread" with the following body:
+      """
+      {
+        "status": "waiting_for_trainer",
+        "message_count": 1,
+        "helper_group_id": 100
+      }
+      """
+    Then the response should be "updated"
+    And the table "threads" should stay unchanged but the row with item_id "1000"
+    And the table "threads" at item_id "1000" should be:
+      | latest_update_at    | message_count | status              | helper_group_id |
+      | 2022-01-01 00:00:00 | 1             | waiting_for_trainer | 100             |
+
+  # To write on a thread, a user must fulfill either of those conditions:
+  #  (1) be the participant of the thread
+  #  (2) have can_watch>=answer permission on the item AND can_watch_members on the participant
+  #  (3) be part of the group the participant has requested help to AND either have can_watch>=answer on the item
+  #    OR have validated the item.
+  Scenario: Can write to thread condition (1) when status is not set
+    Given I am the user with id "1"
+    And There is a thread with "item_id=10,participant_id=1"
+    When I send a PUT request to "/items/10/participant/1/thread" with the following body:
+      """
+      {
+        "message_count": 1
+      }
+      """
+    Then the response should be "updated"
+    And the table "threads" should stay unchanged but the row with item_id "10"
+    And the table "threads" at item_id "10" should be:
+      | latest_update_at    | message_count |
+      | 2022-01-01 00:00:00 | 1             |
+
+  Scenario: Can write to thread condition (2) when status is not set
+    Given I am the user with id "2"
+    And There is a thread with "item_id=20,participant_id=3"
+    And I can watch answer on item with id "20"
+    And I can watch the participant with id "3"
+    When I send a PUT request to "/items/20/participant/3/thread" with the following body:
+      """
+      {
+        "message_count": 2
+      }
+      """
+    Then the response should be "updated"
+    And the table "threads" should stay unchanged but the row with item_id "20"
+    And the table "threads" at item_id "20" should be:
+      | latest_update_at    | message_count |
+      | 2022-01-01 00:00:00 | 2             |
+
+  Scenario: Can write to thread condition (3) with can_watch>=answer on the item, when status is not set
+    Given I am the user with id "2"
+    And There is a thread with "item_id=30,participant_id=3"
```

Comment:
there should be an upper case to "There" ... it should be "And there is ... ". Probably you have been mislead by the "I am ..."